### PR TITLE
feat(webapp): add ssh key authentication

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
demo video:

https://github.com/Devolutions/devolutions-gateway/assets/139169536/6dae002b-2b21-4c1e-8316-61a8bc95ef1a

I tried to use PrimeNG file upload component, but it does not suit our need.
Please let me know if by using extraData attribute a good idea when creating new session.

### CODE REVIEW

File: webapp/src/client/app/shared/services/web-form.service.ts [hunk 1]
```diff
@@ -73,4 +86,16 @@ export class WebFormService extends BaseComponent {
...
+
+  public canConnect(form: FormGroup): boolean {
+    return form.valid && this.canConnectExtraCallback();
+  }
+
+  canConnectIfAlsoTrue(callback: () => boolean): void {
+    this.canConnectExtraCallback = () => callback();
+  }
+
+  resetCanConnectCallback(){
```
Comment (kristahouse) [inline webapp/src/client/app/shared/services/web-form.service.ts:98]: I'll be picky here and ask to add a space after the parathesis 🧐 Everything else looks great!
`resetCanConnectCallback() {`

**Comment by kristahouse** [comment]
Code looks good. Nice job @irvingoujAtDevolution! 

I will add Paul's comments so we can keep @CBenoit in the loop:

Following the UI in DVLS for the ssh key: We usually display the key in a text box, after the user uploads the key. The text box could also allow the user to type the key manually or paste it in quickly.

**Comment by CBenoit** [comment]
> I will add Paul's comments so we can keep @CBenoit in the loop:
> 
> Following the UI in DVLS for the ssh key: We usually display the key in a text box, after the user uploads the key. The text box could also allow the user to type the key manually or paste it in quickly.

Thank you! This is a sensible UX. Code-wise, I have nothing more to add!

**Comment by irvingoujAtDevolution** [comment]
https://github.com/Devolutions/devolutions-gateway/assets/139169536/0e0f1c05-521a-4049-9284-db2913e6049a

updated as described

File: webapp/src/client/app/modules/web-client/form/form-controls/file-control/file-control.component.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            <span>
+                <p-button
+                    type="button"
+                    [label]="'clear'"
+                    (onClick)="clearPublicKeyData()"
+                >
+                </p-button>
+                <p-button
+                    type="button"
+                    [label]="'upload'"
```
Comment (CBenoit) [inline webapp/src/client/app/modules/web-client/form/form-controls/file-control/file-control.component.html:25]:
```suggestion
                    [label]="'Upload'"
```

**Comment by irvingoujAtDevolution** [comment]
> Out of curiosity, is it still possible to drag and drop the file?

No, it is not yet possible.

**Comment by CBenoit** [comment]
Do you think you could check with Paul or Krista if this should be implemented?

**Comment by irvingoujAtDevolution** [comment]
> Do you think you could check with Paul or Krista if this should be implemented?

sure, that's a good idea.

**Comment by irvingoujAtDevolution** [comment]
Uploading GW final versio private key.mp4…

**Pull request merged**